src/plugins/typechecker/README.md

```diff
@@ -0,0 +1,190 @@
+- infos = Information about the typechecker plugin is in keys below
+- infos/author = Armin Wurzinger <e1528532@libelektra.org>
+- infos/licence = BSD
+- infos/needs = 
+- infos/provides = typechecker
+- infos/recommends =
+- infos/placements = postgetstorage presetstorage
+- infos/status = maintained experimental global
+- infos/metadata =
+- infos/description = a plugin which typechecks configuration specifications
+
+## Introduction
+
+A plugin which typechecks specifications before setting keys and after getting keys from
+a mounted specification.
+
+The typesystem is based on regular expressions. Each key is assigned with a regex
+that describes its contents. Links between keys can only be done if the regexes describing 
+the linked keys are compatible with each other, i.e. the regex of the linked key is equal
+or a subset of the regex of the linking key. This concept is not only useful for defining
+links. It can be used to restrict the input types of transformations for 
+instance as well.
+
+## Usage
+
+There are basically two ways to use typechecker plugin. First we describe how the plugin
+is used in general for configuration specification developers and users. Afterwards we 
```

Comment:
"specification developers and users" are for me already two groups. But you can keep this for this PR.